lib/services/role.go

```diff
@@ -1569,6 +1578,14 @@ func (set RoleSet) CheckLoginDuration(ttl time.Duration) ([]string, error) {
 	if !matchedTTL {
 		return nil, trace.AccessDenied("this user cannot request a certificate for %v", ttl)
 	}
+	if len(logins) == 0 && !set.hasPossibleLogins() {
```

Comment:
Can we please add a couple of tests for this edge case to make sure we haven't broken anything. I'm scared to mess this part up, because if we do, the fallout will be of the epic proportions :)